app/data-generation/get-delta-amount.js

```diff
@@ -0,0 +1,24 @@
+const db = require('../data')
+const { getValue } = require('./get-value')
+
+const getDeltaAmount = async (event, transaction) => {
+  const requestNumber = event.data.paymentRequestNumber
+  const value = getValue(event)
+  if (requestNumber === 1 || value === null) {
+    return value
+  }
+  const previousRequest = await db.reportData.findOne({
+    where: {
+      paymentRequestNumber: requestNumber - 1,
+      sourceSystem: event.data.sourceSystem,
+      frn: event.data.frn,
+      agreementNumber: event.data.agreementNumber
```

Comment:
We probably want to rework this somewhat. (my mistake)

This is good for most schemes, but some schemes don't use the agreement number so we should match differently.

Here's how the filter is defined when we do similar checks in _ffc-pay-processing_:

```
  switch (paymentRequest.schemeId) {
    case BPS:
      return {
        ...defaultFilter,
        schemeId: paymentRequest.schemeId,
        frn: paymentRequest.frn,
        marketingYear: paymentRequest.marketingYear
      }
    case FDMR:
      return {
        ...defaultFilter,
        schemeId: paymentRequest.schemeId,
        frn: paymentRequest.frn
      }
    case CS:
      return {
        ...defaultFilter,
        schemeId: paymentRequest.schemeId,
        frn: paymentRequest.frn,
        [db.Sequelize.Op.or]: [
          { contractNumber: paymentRequest.contractNumber },
          db.Sequelize.where(db.Sequelize.fn('replace', db.Sequelize.col('contractNumber'), 'A0', 'A'), paymentRequest.contractNumber?.replace('A0', 'A'))
        ]
      }
    default:
      return {
        ...defaultFilter,
        schemeId: paymentRequest.schemeId,
        frn: paymentRequest.frn,
        marketingYear: paymentRequest.marketingYear,
        agreementNumber: paymentRequest.agreementNumber
      }
  }

```

Where default filter is equivalent to our _paymentRequestNumber_ check.

_sourceSystem_ is doing the same lifting as _schemeId_. So we want to add _marketingYear_ to this default filter, and have the same kind of case driven filter for CS, FDMR and BPS. Our event should have a _event.data.schemeId_, so we can probably do the same kind of case driven that just checks for these values. 

Here are the schemes and their values, suggest we make a new constant file for this.
```
module.exports = {
  SFI: 1,
  SFI_PILOT: 2,
  LUMP_SUMS: 3,
  VET_VISITS: 4,
  CS: 5,
  BPS: 6,
  FDMR: 7,
  MANUAL: 8,
  ES: 9,
  FC: 10,
  IMPS: 11,
  SFI23: 12
}
```